site/components/map/RoadWorksMap.tsx

```diff
@@ -0,0 +1,206 @@
+    },
+};
+
+const clusterCountLayer: LayerProps = {
+    id: "cluster-count",
+    type: "symbol",
+    source: "roadwork-icon-disruptions",
+    filter: ["has", "point_count"],
+    layout: {
+        "text-field": "{point_count_abbreviated}",
+        "text-font": ["DIN Offc Pro Medium", "Arial Unicode MS Bold"],
+        "text-size": 12,
+    },
+};
+
+const roadworkLayer: SymbolLayer = {
+    id: "unclustered-point",
+    type: "symbol",
+    source: "roadwork-icon-disruptions",
+    filter: ["!", ["has", "point_count"]],
+    layout: {
+        "icon-image": "cone",
+        "icon-size": 0.08,
+    },
+};
+
+const Map = ({ initialViewState, style, mapStyle, roadworks }: MapProps): ReactElement | null => {
+    const mapboxAccessToken = process.env.MAP_BOX_ACCESS_TOKEN;
+
+    const mapRef = useRef<MapRef>(null);
+
+    const [popupInfo, setPopupInfo] = useState<
+        Partial<{
+            streetName: string;
+            startDateTime: string;
+            endDateTime: string;
+            longitude: number;
+            latitude: number;
+            permitReferenceNumber: string;
+        }>
+    >({});
+
+    const roadworkSource: FeatureCollection = {
+        type: "FeatureCollection",
+        features: roadworks.map((r) => ({
+            ...r.worksLocationCoordinates,
+            properties: {
+                streetName: startCase(lowerCase(r.streetName || "")),
+                startDateTime: r.actualStartDateTime
+                    ? convertDateTimeToFormat(r.actualStartDateTime, "DD/MM/YYYY HHmm")
+                    : r.proposedStartDateTime
+                    ? convertDateTimeToFormat(r.proposedStartDateTime, "DD/MM/YYYY HHmm")
+                    : "",
+                endDateTime: r.actualEndDateTime
+                    ? convertDateTimeToFormat(r.actualEndDateTime, "DD/MM/YYYY HHmm")
+                    : r.proposedEndDateTime
+                    ? convertDateTimeToFormat(r.proposedEndDateTime, "DD/MM/YYYY HHmm")
+                    : "",
+                longitude: r.worksLocationCoordinates.geometry.coordinates[0],
+                latitude: r.worksLocationCoordinates.geometry.coordinates[1],
+                permitReferenceNumber: r.permitReferenceNumber,
+            },
+        })),
+    };
+
+    const MapImage = () => {
+        const { current: map } = useMap();
+
+        if (map) {
+            if (!map.hasImage("cone")) {
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-argument,@typescript-eslint/no-unsafe-member-access
+                map.loadImage(cone.src, (error, image) => {
+                    if (error) throw error;
+                    if (!map.hasImage("cone") && image) map.addImage("cone", image, { sdf: false });
+                });
+            }
+        }
+
+        return null;
+    };
+
+    const onClick = useCallback((event: MapLayerMouseEvent) => {
+        const feature = event.features?.[0];
```

Comment:
Could we map feature variable to type `MapboxGeoJSONFeature` which might allow us to remove some type assertions below

Reply:
this change causes errors below